src/mem3/src/mem3_shards.erl

```diff
@@ -210,12 +216,25 @@ handle_cast({cache_hit, DbName}, St) ->
     couch_stats:increment_counter([mem3, shard_cache, hit]),
     cache_hit(DbName),
     {noreply, St};
-handle_cast({cache_insert, DbName, Shards}, St) ->
-    couch_stats:increment_counter([mem3, shard_cache, miss]),
-    {noreply, cache_free(cache_insert(St, DbName, Shards))};
+handle_cast({cache_insert, DbName, Writer, UpdateSeq}, St) ->
+    NewSt = case UpdateSeq < St#st.update_seq of
```

Comment:
I'll add a comment to explain that given that its caused more than one developer to question it